pandas/tests/window/test_rolling.py

```diff
@@ -1734,3 +1735,38 @@ def test_rolling_std_neg_sqrt():
 
     b = a.ewm(span=3).std()
     assert np.isfinite(b[2:]).all()
+
+
+def test_rolling_aggregation_boundary_consistency(rolling_aggregation):
+    # GH-45647
+    minp, step, width, size, selection = 0, 1, 3, 11, [2, 7]
+    s = Series(np.arange(1, 1 + size, dtype=np.float64))
+    end = np.arange(width, size, step, dtype=np.int64)
+    start = end - width
+    selarr = np.array(selection, dtype=np.int32)
+    result = Series(rolling_aggregation(s.values, start[selarr], end[selarr], minp))
```

Comment:
Typically it's better to use `to_numpy()` instead of `.values`

Reply:
I will simplify the code by using an array directly.